### ISSUE TITLE
getNumberOfEvents changes the internal state of the file

### ISSUE THREAD

**Issue by jstrube** [iLCSoft/LCIO#65]
!!!IMPORTANT!!!: To facilitate faster and easier response to your issue please provide in addition to the description of the issue also the following information

 - OS version: SLC6, Ubuntu 18.04
 - Compiler version: GCC 6.2, gcc 7.4
 - Package version: v02-12-01
 - Reproduced by: call `readNextEvent()`, followed by `getNumberOfEvents`, followed by `readNextEvent()`. You have just read the same event twice! 
 - Goal: It is not obvious that `getNumberOfEvents` leads to this behavior, nor should it. The name suggest that this is a getter method. However, it looks like the implementation changes the internal state of the file.
Tested on `/pnfs/desy.de/ilc/prod/ilc/mc-opt-3/ild/dst-merged/500-TDR_ws/2f_Z_hadronic/ILD_l5_o1_v02/v02-00-01/rv02-00-01.sv02-00-01.mILD_l5_o1_v02.E500-TDR_ws.I250114.P2f_z_h.eL.pR.n033.d_dstm_10410_96.slcio`

**Comment by rete** [comment]
Thanks for reporting this.
Indeed, calling `getNumberOfEvents()` is doing the following:
- ask the random access manager to read the index at the end of the file
- populate the event map
- **reset the stream pointer to begin**
This is why you read twice the same event I guess

**Comment by gaede** [comment]
I agree that this seems not obvious - should change the documentation.
